docs/deploy.md

```diff
@@ -0,0 +1,205 @@
+
+Tock is deployed to the GovCloud instance of cloud.gov. You will need to login
+to via the GovCloud api of cloud.gov:
+`cf login -a api.fr.cloud.gov --sso`
+
+Then target the org and space you want to work with. For example, if you wanted to work with the dev space:
+`cf target -o gsa-18f-tock -s dev`
+
+Manifest files, which contain import deploy configuration settings, are located
+in the root directory of this project, prefixed with `manifest-` and ending in a
+`.yml` file extension.
+
+Note that this project has two requirements files:
+* `requirements.txt` for production dependencies
+* `requirements-dev.txt` for development and testing dependencies
+
+During local development and continuous integration testing,
+`pip install -r requirements-dev.txt` is used, which installs both
+development and production dependencies. During deployments, the Cloud
+Foundry python buildpack uses only `requirements.txt` by default, so
+only production dependencies will be installed.
+
+### Cloud Foundry structure
+
+- cloud.gov environment: `GovCloud`
+- Organization: `gsa-18f-tock`
+- Spaces: `staging`, `prod`
+- Apps:
+  - `staging` space:
+    - `tock-staging`
+  - `prod` space:
+    - `tock`
+- Routes:
+  - tock.app.cloud.gov -> `staging` space, `tock-staging` app
+  - calc-staging.app.cloud.gov -> `staging` space, `calc-staging` app
+  - tock.18f.gov -> `prod` space, `tock` app
+
+### Services
+
+#### User Provided Service (UPS)
+tockdb                            aws-rds                     shared-psql      tock         update succeeded
+tockdb-new                        aws-rds                     shared-psql                   delete failed
+tock-creds-prod                   user-provided                                tock
+
+For cloud.gov deployments, this project makes use of a [User Provided Service (UPS)][UPS] to get its configuration
+variables, instead of using the local environment (except for [New Relic-related environment variables](#new-relic-environment-variables)).
+You will need to create a UPS called `tock-credentials`, provide 'credentials' to it, and link it to the
+application instance. This will need to be done for every Cloud Foundry `space`.
+
+<!--
+First, create a JSON file (e.g. `credentials-staging.json`) with all the configuration values specified as per the
+[Environment variables](environment.md). **DO NOT COMMIT THIS FILE.**
+
+```json
+{
+  "SECRET_KEY": "my secret key",
+  "...": "other environment variables"
+}
+```
+-->
+
+Then enter the following commands (filling in the main application instance name
+for `<APP_INSTANCE>`) to create the user-provided service:
+
+```sh
+cf cups tock-credentials -p credentials-<ENVIRONMENT>.json
+cf bind-service <APP_INSTANCE> tock-credentials
+cf restage <APP_INSTANCE>
+```
+
+You can update the user-provided service with the following commands:
+
+```sh
+cf uups tock-credentials -p credentials-staging.json
+cf restage <APP_INSTANCE>
```

Comment:
For zero-down time, we should suggest also redeploying the app, e.g., rebuilding the last production build in CircleCI.